vpr/src/route/check_route.cpp

```diff
@@ -325,18 +325,20 @@ static bool check_adjacent(int from_node, int to_node) {
 
     num_adj = 0;
 
-    from_type = rr_graph.node_type(RRNodeId(from_node));
-    from_xlow = rr_graph.node_xlow(RRNodeId(from_node));
-    from_ylow = rr_graph.node_ylow(RRNodeId(from_node));
-    from_xhigh = rr_graph.node_xhigh(RRNodeId(from_node));
-    from_yhigh = rr_graph.node_yhigh(RRNodeId(from_node));
-    from_ptc = device_ctx.rr_nodes[from_node].ptc_num();
-    to_type = rr_graph.node_type(RRNodeId(to_node));
-    to_xlow = rr_graph.node_xlow(RRNodeId(to_node));
-    to_ylow = rr_graph.node_ylow(RRNodeId(to_node));
-    to_xhigh = rr_graph.node_xhigh(RRNodeId(to_node));
-    to_yhigh = rr_graph.node_yhigh(RRNodeId(to_node));
-    to_ptc = device_ctx.rr_nodes[to_node].ptc_num();
+    auto from_rr = RRNodeId(from_node);
+    auto to_rr = RRNodeId(to_node);
+    from_type = rr_graph.node_type(from_rr);
+    from_xlow = rr_graph.node_xlow(from_rr);
+    from_ylow = rr_graph.node_ylow(from_rr);
+    from_xhigh = rr_graph.node_xhigh(from_rr);
+    from_yhigh = rr_graph.node_yhigh(from_rr);
+    from_ptc = rr_graph.node_ptc_num(from_rr);
+    to_type = rr_graph.node_type(to_rr);
+    to_xlow = rr_graph.node_xlow(to_rr);
+    to_ylow = rr_graph.node_ylow(to_rr);
+    to_xhigh = rr_graph.node_xhigh(to_rr);
+    to_yhigh = rr_graph.node_yhigh(to_rr);
+    to_ptc = rr_graph.node_ptc_num(to_rr);
```

Comment:
I read the codes in this function. It seems that the ``to_ptc`` is only used by SOURCE and IPIN nodes. Should we use specific API in these cases rather than depositing here? I do not have a strong opinion here. Your call.

Reply:
I think the code is fine this way. We could move the call lower into the individual case statements, thought if you think that would be better.

Reply:
No worries. Let's keep it in this way.